pkg/splunk/enterprise/workerpool.go

```diff
@@ -0,0 +1,184 @@
+// Copyright (c) 2018-2021 Splunk Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package enterprise
+
+import (
+	"fmt"
+	"io/fs"
+	"os"
+	"sync"
+
+	enterpriseApi "github.com/splunk/splunk-operator/pkg/apis/enterprise/v2"
+	splclient "github.com/splunk/splunk-operator/pkg/splunk/client"
+)
+
+// AppInstallWorkerPool is a pool of workers to install apps in parallel
+type AppInstallWorkerPool struct {
```

Comment:
Nice work!  Will this struct be able to handle all parameters needed to download and install apps for a given CR?  Will the pool be used across CRs or per CR?

Reply:
This is per CR

Reply:
Good point .. wonder if we need a pool of pools to avoid OOM etc?

Reply:
I am sorry but did not understand the concerns here. @jryb @smohan-splunk ?

Reply:
I didn't have a particular concern just wondering if each pool is per CR or per operator instance.  If it's per CR then is there a concern that there could be too many threads open for operators that are managing a number of CRs?  My guess would be that this likely wouldn't be an issue of running out of threads or memory, but might be worth a thought.